hackforla/peopledepot
# switch to alpine image

Fixes #177

### What changes did you make?

- switched docker build to start from `alpine` base image
- added `Roboto` font
- modified `erd.sh` script to produce the same result as before without `pydot`
- removed some packages that used to be installed by apt

### Why did you make the changes (we will use this info to test)?

- Switching to `alpine` is to save space, download time, and build time
- alpine doesn't come with fonts so I installed the Roboto font which is hard coded by the `graph_models` command
- The `graph_models` command using `pydot` doesn't produce the correct output anymore, so I split the operation into 2 commands in the `erd.sh` script and removed `pydot` as a dependency
- Other python package dependencies were removed here. I suspect they may have been generated by `uv` in an earlier PR (#268) where it detected my system `python` to be `3.12`. This time I made sure to generate the `requirements.txt` using the `python 3.10` in the docker image.
  - The strange thing is I noticed the extra additions at the time of that older PR and tried without success to generate one that didn't have them, and running against the python in the image was something I thought about. I even added a script to call it, which I'm sure I ran at the time. That's the script I ran this time.
- Some of the apt-installed packages are not needed by django or erd-generation.
  - postgres doesn't seem to be required according to several team members
  - I remember the older graphviz or pydot version required gcc, which is no longer the case
  - netcat was used in the database wait script, which we removed

### Testing suggestions

- make sure it builds and the tests still pass
- run `scripts/erd.sh` and make sure it still generates `app/erd.png`, and running `file app/erd.png` from the host system shows a normal image and not something like `1px x 800000px`

    ```
    ❯ file erd.png 
    erd.png: PNG image data, 4391 x 1396, 8-bit/color RGBA, non-interlaced
    ```
- check that django admin looks okay
